src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,681 @@
+        eachProductArea: ".pmt-single-product-area",
+        topbarSaveButton: "#pmt-topbar-save-button", // Button on topbar to open Preview/Confirm Save Dialog
+        devModeIcon: "#pmt-topbar-devmode-button",
+        deleteContextButtons: ".pmt-delete-context"
+    },
+    templates: {
+        initial: "editprefset-viewport",
+        editPrefWidget: "editpref-widget"
+    },
+    invokers: {
+        reRender: {
+            func: "{that}.events.refresh.fire"
+        },
+        addEditToUnsavedList: {
+            funcName: "gpii.devpmt.addEditToUnsavedList",
+            args: ["{that}", "{arguments}.0"]
+        },
+        commonTermUsageCounts: {
+            funcName: "gpii.devpmt.commonTermUsageCounts",
+            args: ["{that}", "{that}.model.commonTermsSorted", "{that}.model.flatPrefs"]
+        },
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        renderEditSidebar: {
+            func: "{that}.renderMarkup",
+            args: ["editWidgetSidebar", "{that}.options.templates.editPrefWidget", "{arguments}.0"]
+        },
+        editValueEvent: {
+            funcName: "gpii.devpmt.editValueEvent",
+            args: ["{that}", "{arguments}.0"]
+        },
+        editProductEnabled: {
+            funcName: "gpii.devpmt.editProductEnabled",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+        },
+        savePrefset: {
+            funcName: "gpii.devpmt.savePrefset",
+            args: ["{that}" /*, "{arguments}.0" */]
+        },
+        updateMetadataFromPrefs: {
+            funcName: "gpii.devpmt.updateMetadataFromPrefs",
+            args: ["{that}"]
+        },
+        initSettingTableWidgets: {
+            funcName: "gpii.devpmt.initSettingTableWidgets",
+            args: ["{that}"]
+        },
+        openAddProductDialog: {
+            funcName: "gpii.devpmt.openAddProductDialog",
+            args: ["{that}", "{arguments}.0"] // appId
+        },
+        toggleDevModeView: {
+            funcName: "gpii.devpmt.toggleDevModeView",
+            args: ["{that}", "{that}.model.devModeOn"]
+        },
+        lookupGenericPrefValue: {
+            funcName: "gpii.devpmt.lookupGenericPrefValue",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"] // context, commonTerm
+        },
+        lookupProductPrefValue: {
+            funcName: "gpii.devpmt.lookupProductPrefValue",
+            args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // context, product, settingTerm
+        },
+        onDeleteContext: {
+            funcName: "gpii.devpmt.onDeleteContext",
+            args: ["{that}", "{arguments}.0"] // event
+        }
+    },
+    listeners: {
+        "onCreate": {
+            funcName: "gpii.devpmt.npsetInit",
+            args: ["{that}"] //"onCreate listener"]
+        },
+        "onMarkupRendered": [
```

Comment:
There's a lot of manual "click" stuff across the components. You probably want to i) consolidate it into some piece of micro-framework such as https://github.com/fluid-project/fluid-authoring/blob/FLUID-4884/src/js/DecoratorViewComponent.js#L15 (and propose something similar in the end to go upstairs into gpii-binder), ii) put the definition into a namespaced/prioritied listener to onMarkupRendered in a base grade, iii) namespace/priorities the last two entries here, updateFoundationSticky and initSettingTableWidgets

Reply:
These have all been replaced with a `gpii.binder.binderMarkupEvent` component, and noted in the following ticket: GPII-2933